byteparse/shared/src/main/scala/byteparse/classparse/ClassAttributes.scala

```diff
@@ -0,0 +1,183 @@
+package byteparse
+package classparse
+
+import fastparse.allByte._
+
+object ClassAttributes {
+  import ClassParser.Info._
+  import ClassParser.BasicElems._
+  import ClassParser.Ast._
+  import ClassParser._
+  import CodeParser._
+
+  sealed abstract class Attribute
+
+  case class BasicAttribute(name: String, info: Array[Byte]) extends Attribute {
+    override def equals(other: Any): Boolean =
+      other match {
+        case attr: BasicAttribute => name == attr.name && info.deep == attr.info.deep
+        case _ => false
+      }
+
+    override def toString = {
+      import fastparse.ElemTypeFormatter.ByteFormatter
+      s"BasicAttribute($name,${ByteFormatter.prettyPrint(info)})"
+    }
+  }
+
+  case class ConstantValueAttribute(value: BasicElem) extends Attribute
+
+  case class ExceptionHandler(startIdx: Int, endIdx: Int, handlerIdx: Int, catchType: Option[Class])
+
+  case class CodeAttribute(maxStack: Int, maxLocals: Int,
+                           code: Seq[OpCode], exceptions: Seq[ExceptionHandler],
+                           attributes: Seq[Attribute]) extends Attribute
+
+  case class ExceptionsAttribute(exceptions: Seq[Class]) extends Attribute
+
+
+  case class InnerClass(innerClass: Class, outerClass: Option[Class],
+                        innerName: Option[String], accessFlags: InnerClassFlags)
+
+  case class InnerClassesAttribute(classes: Seq[InnerClass]) extends Attribute
+
+  case class EnclosingMethodAttribute(enclosingClass: Class, enclosingMethodName: Option[String],
+                                      enclosingMethodDescriptor: Option[String]) extends Attribute
+
+  case class SyntheticAttribute() extends Attribute
+
+  case class SignatureAttribute(signature: String) extends Attribute
+
+  case class SourceFileAttribute(sourceFile: String) extends Attribute
+
+  case class DeprecatedAttribute() extends Attribute
+
+  case class BootstrapMethod(bootstrapMethodRef: MethodHandle, bootstrapArguments: Seq[PoolItem])
+
+  case class BootstrapMethodsAttribute(bootstrapMethods: Seq[BootstrapMethod]) extends Attribute
+
+  import ByteUtils.BE._
+
+  val attributeParsers = Map[String, (ClassFileInfo) => Parser[Attribute]](
+    "ConstantValue" -> {
+      (classInfo: ClassFileInfo) =>
+        P( AnyWordI ).map(idx =>
+          ConstantValueAttribute(
+            classInfo.getInfoByIndex[PoolInfo](idx).get match {
+              case BasicElemInfo(elem) => elem
+              case StringInfo(idx) => StringElem(classInfo.getStringByIndex(idx))
+            }))
+    },
+    "Code" -> {
+      (classInfo: ClassFileInfo) => {
+        val exceptionHandler =
```

Comment:
Did you push your changes? I'm looking on github and it `exceptionHandler` still looks like
```scala
+  val code = {
 +      val exceptionHandler =
 +        P( AnyWordI /*start_pc*/ ~
 +           AnyWordI /*end_pc*/ ~
 +           AnyWordI /*handler_pc*/ ~
 +           AnyWordI /*catch_type*/
 +        ).map{
 +          case (spc: Int, epc: Int, hpc: Int, ctype: Int) =>
 +            (classInfo: ClassFileInfo) => ExceptionHandler(
 +              spc,
 +              epc,
 +              hpc,
 +              if (ctype == 0) None
 +              else Some(Class(classInfo, classInfo.getInfoByIndex[ClassInfo](ctype).get)))
 +        }
```

And many (most?) of the other parsers look the same, still spread over multiple lines with `/**/` comments on each terminal rather than assigning them to names